### DISCUSSION TITLE
What are long-term goals for CKAN?

### DISCUSSION THREAD

**Discussion by thegostev** [ckan/ckan#7479] [CKAN 3.0]
Hi.

Defining desired outcomes is the vital step we should take to agree on where CKAN should be in the future.

We must define what success would look like in order to know that we achieved it. Skipping this step will lead us into discussing solutions. Solutions are fine but how to prioritize them if we haven't set up our desired goals. Additionally, solutions that have strong connections to desired outcomes worth more attention, aren't they?

**What goals do you believe we should pursue for CKAN?**

Yearly goals I can think about:
- CKAN has great UX for publishers and users (so they can do their job faster and with greater satisfaction)
- Increase size of active community participants (so we increase amount of contributions)
- Increase reach of CKAN (so more people around the world know how CKAN can solve their problems and support them)
- Increase amount of systems CKAN can interact with
- Increase usage of CKAN via API
- Make CKAN all-in-one solution for EU governments on a national and local levels
- Make CKAN satisfy requirements of become a Data Hub for everything governments and non-profits need around open data

Below👇 is the map of CKAN Ecosystem to consult with on what components are there so they could change in future. What's missing from this map are technical components of CKAN architecture (and specific extensions) that can also be a subject for improvement.

<img width="730" alt="Screenshot 2023-04-04 at 12 03 47" src="https://user-images.githubusercontent.com/25153729/229758717-e2a91db3-9d21-4690-9011-3c26f5ee67c9.png">

**Comment by pdelboca** [comment]
I think one key long term goal should be to have 2 or 3 full time developers assigned to the project.

I don't think one long term goal should be to increase the amount of contributions. Currently we do not have time to review all the contributions we receive (We received/created 50 PRs in the last month, 29 are still open). That is a clear sign that we have a thriving community and lack of resources to address all the necessities.

**Reply by thegostev** [reply]
Thank you, @pdelboca it's the nice point that that our bottleneck is in processing changes. So we should think through how to fix it.

How would you formulate the outcomes for adding developers to the team: What outcomes for the project 2 or 3 full-time developers will produce? Or how they would change CKAN future? What is missing now that these new team members will add?

**Reply by pdelboca** [reply]
Currently all the effort of the Tech Team goes to maintenance. We do not have time to propose new features, innovate or proactively think on upgrades to the architecture, technology, etc.

A couple of full time developers can:
 - Provide capacity to the project to tackle maintenance tasks.
 - Train long term users to replace current Tech Team members in the future.
 - Provide new and fresh ideas to the project
 - Allow the Tech Team to focus on more critical tasks and innovation rather than maintenance.
 - Provide visibility and better cadence to the project release since tasks will no longer depends on "free time" of the Tech Team members.
 - A more manageable backlog since we would have people with fixed allocated time to do things.
 - Add stability to the project.

**Comment by pdelboca** [comment]
Regarding the CKAN Ecosystem Chart, I think a big buggle is missing: Self hosted instances.

CKAN is not only used by Consultants and Vendors but also by a lot of small-teams, governments and organizations. CKAN is a Framework that shouldn't depend explicitly on vendors/consultants. Self managed Tech Teams inside Governments or Organizations should also be added to the list since is a big part of our ecosystem.

**Reply by thegostev** [reply]
Perfect! Thank you. I'm updating the chart.

**Reply by thegostev** [reply]
@pdelboca please review.

**Reply by pdelboca** [reply]
@thegostev thanks for the update! I like the new version!

**Reply by tino097** [reply]
we need to find a way to include those smaller teams into community and encourage them to get more involved into activities.
